### PR TITLE
backend/feat: add location db dhall configs

### PR DESCRIPTION
**Pull request by witcher-shailesh**
Dependency PR shared-kernel :- https://github.com/nammayatri/shared-kernel/pull/140

## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverLocation.hs [hunk 1]
```diff
@@ -106,12 +107,12 @@ updateOnRide driverId onRide merchantId = do
...
+updateOnRideCache :: (CacheFlow m r, Esq.EsqDBFlow m r, EsqLocDBFlow m r) => Id Person.Driver -> m ()
 updateOnRideCache driverId = do
   driverLocation <- DLQueries.findById (cast driverId) >>= fromMaybeM (PersonNotFound driverId.getId)
   cacheDriverLocation driverLocation
 
-updateOnRideCacheForCancelledOrEndRide :: (CacheFlow m r, EsqDBReplicaFlow m r, EsqDBFlow m r) => Id Person.Driver -> Id Merchant -> m ()
+updateOnRideCacheForCancelledOrEndRide :: (CacheFlow m r, EsqDBReplicaFlow m r, EsqDBFlow m r, EsqLocDBFlow m r) => Id Person.Driver -> Id Merchant -> m ()
 updateOnRideCacheForCancelledOrEndRide driverId merchantId = do
   mbDriverLocation <- findById merchantId (cast driverId)
   maybe
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverLocation.hs:118]: `DLQueries.upsertGpsCoord` needs to run in location DB?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs [hunk 2]
```diff
@@ -167,7 +167,7 @@ rideInfo merchantShortId reqRideId = do
 
   riderId <- booking.riderId & fromMaybeM (BookingFieldNotPresent "rider_id")
   riderDetails <- runInReplica $ QRiderDetails.findById riderId >>= fromMaybeM (RiderDetailsNotFound rideId.getId)
-  mDriverLocation <- runInReplica $ QDrLoc.findById driverId
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs:170]: This is required here?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs [hunk 3]
```diff
@@ -267,7 +267,7 @@ driverLocation :: ShortId DM.Merchant -> Maybe Int -> Maybe Int -> Common.Driver
 driverLocation merchantShortId mbLimit mbOffset req = do
   merchant <- findMerchantByShortId merchantShortId
   let driverIds = coerce req.driverIds
-  allDrivers <- Esq.runInReplica $ QPerson.findAllDriversByIdsFirstNameAsc merchant.id driverIds
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs:270]: `runInReplica` required here as well?

File: Backend/test/src/Mobility/ARDU/NearestDrivers.hs [hunk 4]
```diff
@@ -102,10 +100,3 @@ setDriversActive :: Bool -> Maybe DI.DriverMode -> FlowR ARDUEnv.AppEnv ()
 setDriversActive isActive mode = Esq.runTransaction $ do
   let drivers = [furthestDriver, closestDriver, suvDriver, sedanDriver, hatchbackDriver, driverWithOldLocation]
   forM_ drivers (\driver -> Q.updateActivity (Id driver) isActive mode)
-
--- we can remove this when we flatten migrations
-setDriverWithOldLocation :: FlowR ARDUEnv.AppEnv ()
```
Comment (NikithShetty) [inline Backend/test/src/Mobility/ARDU/NearestDrivers.hs:107]: Is this used in any test case?

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverLocation.hs (hunk 1 above)
Comment (witcher-shailesh) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverLocation.hs:118]: Done

**Review by srid — APPROVED** [review]

**Review by NikithShetty — APPROVED** [review]

**Review by NikithShetty — APPROVED** [review]

**Pull request merged**